File: src/helpers/faucet/faucet-requester.ts
/* eslint-disable @typescript-eslint/restrict-plus-operands */
/* eslint-disable @typescript-eslint/restrict-template-expressions */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import axios from 'axios';

import { Config } from '../../../config/default';
import { logger } from '../../utils/logger';
import { FaucetRequest } from './faucet-request';

export const requestFaucet = async (wallet: string, amount: number): Promise<void> => {
  if (!Config.useFaucet) {
    logger.notice('Skipping faucet request');
    return;
  }
  const data: FaucetRequest = { amount: amount, wallet: wallet };
  logger.notice(`URL: ${Config.faucetUrl}`);
  logger.notice(`Wallet = ${data.wallet}, amount = ${data.amount}`);
  const result = await axios.post(Config.faucetUrl, data);
  logger.notice(result);
};

File: src/helpers/faucet/faucet_requester.py
import httpx
from config.config import FAUCET_URL, USE_FAUCET


async def request_faucet(wallet: str, amount: int):
    if not USE_FAUCET or FAUCET_URL == '':
        print("Skipping faucet request")
        return
    data = {'amount': int(amount), 'wallet': wallet}
    try:
        async with httpx.AsyncClient() as client:
            response = await client.post(FAUCET_URL, json=data, timeout=20)
            assert response.status_code == 200, "The response status is not OK"
    except Exception as e:
        print(e)
        print(f"Failed to obtain tokens from faucet {FAUCET_URL}")
